stored_requests/events/postgres/database.go

```diff
@@ -0,0 +1,180 @@
+package postgres
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"encoding/json"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/stored_requests/events"
+	"github.com/prebid/prebid-server/util/timeutil"
+)
+
+func bytesNull() []byte {
+	return []byte{'n', 'u', 'l', 'l'}
+}
+
+type PostgresEventProducerConfig struct {
+	DB                 *sql.DB
+	RequestType        config.DataType
+	CacheInitQuery     string
+	CacheInitTimeout   time.Duration
+	CacheUpdateQuery   string
+	CacheUpdateTimeout time.Duration
+}
+
+type PostgresEventProducer struct {
+	cfg           PostgresEventProducerConfig
+	lastUpdate    time.Time
+	invalidations chan events.Invalidation
+	saves         chan events.Save
+	time          timeutil.Time
+}
+
+func NewPostgresEventProducer(cfg PostgresEventProducerConfig) (eventProducer *PostgresEventProducer) {
+	if cfg.DB == nil {
+		glog.Fatalf("The Postgres Stored %s Loader needs a database connection to work.", cfg.RequestType)
+	}
+
+	return &PostgresEventProducer{
+		cfg:           cfg,
+		lastUpdate:    time.Time{},
+		saves:         make(chan events.Save, 1),
+		invalidations: make(chan events.Invalidation, 1),
+		time:          &timeutil.RealTime{},
+	}
+}
+
+func (e *PostgresEventProducer) Run() error {
+	if e.lastUpdate.IsZero() {
```

Comment:
The older version of this code checked for `saves != nil` and `invalidations != nil`, which is still possible - but only if we code it up wrong. I'm fine leaving this validation out, but just wanted to mention its still technically possible since I'm just a bit on the fence.

Reply:
We chatted as a group and decided to leave the code as is.